src/Indigo/Contracts/Safelist.hs

```diff
@@ -2,193 +2,71 @@
-  doc $ DDescription
-    "This contract implements regulatory service that can be used by other contracts."
+safelistIndigo param = contractName "Dummy safelist" $ do
   entryCase (Proxy @PlainEntryPointsKind) param
-    ( #cAddAdmin /-> \addr -> do
-        doc $ DDescription
-          "Add new admin, if added address is already admin, then nothing happens."
-        ensureSenderIsOwner
-        addOrRemoveFromStorageSet addr #sAdmins Insert
-    , #cRemoveAdmin /-> \addr -> do
-        doc $ DDescription
-          "Remove admin, address is not an admin already, then nothing happens."
-        ensureSenderIsOwner
-        addOrRemoveFromStorageSet addr #sAdmins Remove
-    , #cIsAdmin /-> \v -> do
-        doc $ DDescription
-          "Check whether address is admin. Returns `True` if the address is admin."
-        presentedInStorageSet v #sAdmins
-    , #cAddToWhitelist /-> \addr -> do
-        doc $ DDescription
-          "Add address to whitelist, if added address is already there, then \
-          \nothing happens."
-        ensureSenderIsAdmin
-        addOrRemoveFromStorageSet addr #sWhitelist Insert
-    , #cRemoveFromWhitelist /-> \addr -> do
-        doc $ DDescription
-          "Remove address from whitelist, if added address is already not there, \
-          \then nothing happens."
-        ensureSenderIsAdmin
-        addOrRemoveFromStorageSet addr #sWhitelist Remove
-    , #cIsWhitelisted /-> \v -> do
-        doc $ DDescription
-          "Check whether address is whitelisted. Returns `True` if the address is \
-          \whitelisted."
-        presentedInStorageSet v #sWhitelist
-    , #cAddToBlacklist /-> \addr -> do
-        doc $ DDescription
-          "Add address to blacklist, if added address is already there, then \
-          \nothing happens."
-        ensureSenderIsAdmin
-        addOrRemoveFromStorageSet addr #sBlacklist Insert
-    , #cRemoveFromBlacklist /-> \addr -> do
-        doc $ DDescription
-          "Remove address from blacklist, if added address is already not there, \
-          \then nothing happens."
-        ensureSenderIsAdmin
-        addOrRemoveFromStorageSet addr #sBlacklist Remove
-    , #cIsBlacklisted /-> \v -> do
-        doc $ DDescription
-          "Check whether address is blacklisted. Returns `True` if the address is \
-          \blacklisted."
-        presentedInStorageSet v #sBlacklist
-    , #cEnsureSafelistConstraints /-> \constraints -> do
-        doc $ DDescription $
-          "Check list of constraints. For the first list check that all addresses are \
-          \whitelisted, for the second list check that all addresses are non-blacklisted. \
-          \If all constraints are satisfied, nothing happens, otherwise, it will fail \
-          \on the first unsatisfied constraint."
-        ensureSafelistConstraints constraints
+    ( #cAssertTransfer //-> \transfer -> do
+        let fromAddr = UnName #from $ fst transfer
```

Comment:
You could have more simply used `!.` both here and in the next line, e.g. `transfer !. #from`.